argilla/src/argilla/_models/_record/_record.py

```diff
@@ -70,3 +82,25 @@ def validate_metadata(cls, metadata: Union[List[MetadataModel], dict]) -> List[M
         if isinstance(metadata, dict):
             return [MetadataModel(name=key, value=value) for key, value in metadata.items()]
         return metadata
+
+    @field_validator("fields", mode="before")
+    @classmethod
+    def validate_fields(cls, fields: Dict[str, Union[str, None, List[Dict[str, str]]]]) -> Dict[str, FieldValue]:
+        """Ensure fields are a dictionary of field names and values."""
+
+        validated_fields = {}
+        for field_name, field_value in fields.items():
+            if isinstance(field_value, list) and all(isinstance(message, dict) for message in field_value):
+                validated_chat_field_values = []
+                for message in field_value:
+                    if "role" not in message or "content" not in message:
+                        raise ValueError("Chat field values must contain 'role' and 'content' keys.")
+                    if not all(key in ["role", "content"] for key in message.keys()):
+                        warnings.warn(
+                            "Chat field values should only contain 'role' and 'content' keys. Other keys will be ignored by Argilla."
+                        )
+                        message = {key: value for key, value in message.items() if key in ["role", "content"]}
+                        validated_chat_field_values.append(ChatFieldValue(**message))
+                field_value = validated_chat_field_values
+            validated_fields[field_name] = field_value
+        return validated_fields
```

Comment:
This part is a little bit difficult to read. I would try to do some refactor, maybe moving things to other validations or functions?